src/replayer.py

```diff
@@ -0,0 +1,306 @@
+        f"https://{args.api_hostname}/ucfs-claimant/v2/getAwardDetails",
+        data=request_parameters,
+        auth=auth,
+        headers=headers,
+    )
+
+    logger.info(
+        f'Received response from AWS API", "api_hostname": "{args.hostname}", "response_code": "{request.status_code}'
+    )
+
+    return json.load(StringIO(request.text))
+
+
+def decrypt_response(response: dict, request: dict) -> dict:
+    # Create a deep copy of the response to keep the function pure
+    response = response.copy()
+    session = boto3.session.Session(profile_name="decrypt", region_name=region)
+
+    client = session.client("kms")
+
+    for period in response.get("assessmentPeriod", []):
+        amount = period.get("amount")
+
+        key_id = amount.get("keyId")
+        take_home_pay = base64.urlsafe_b64decode(amount.get("takeHomePay"))
+        cipher_text_blob = base64.urlsafe_b64decode(amount.get("cipherTextBlob"))
+
+        kms_response = client.decrypt(CiphertextBlob=cipher_text_blob, KeyId=key_id)
+        data_key = kms_response.get("Plaintext")
+
+        nonce_size = 12
+        # Takes the first 12 characters from the take_home_pay string
+        nonce = take_home_pay[:nonce_size]
+
+        # Takes the remaining characters from the take_home_pay string following the first 12
+        take_home_pay = take_home_pay[nonce_size:]
+
+        aesgcm = AESGCM(data_key)
+
+        try:
+            logger.info(
+                f'Beginning to decrypt data", '
+                f'"transaction_id": {request.get("transaction_id")}, '
+                f'"from_date": {request.get("from_date")}, '
+                f'"to_date": {request.get("to_date")}'
+            )
+            take_home_pay = aesgcm.decrypt(nonce, take_home_pay, None).decode("utf-8")
+
+            amount["takeHomePay"] = take_home_pay
+            del amount["ciperTextBlob"]
+            del amount["keyId"]
+
+            period["amount"] = amount
+
+        except Exception as e:
+            logger.error(
+                f'Failed to decrypt data", '
+                f'"transaction_id": {request.get("transaction_id")}, '
+                f'"from_date": {request.get("from_date")}, '
+                f'"to_date": {request.get("to_date")}'
+            )
+            logger.error(e)
+
+    # Will return a copy of the response if claimantFound is False
+    # Will return a DECRYPTED copy of the response if claimantFound is True
+    return response
+
+
+def compare_responses(original, actual, request):
+    match = True
+    if not (original["claimantFound"] is True and actual["claimantFound"] is True):
+        match = False
+        logger.error(
+            f"Claimant found doesn't match, "
+            f'expected {original["claimantFound"]} from replayed response but got {actual["claimantFound"]}'
+        )
+
+    if original.get("suspendedDate"):
+        state = original["suspendedDate"] == actual["suspendedDate"]
+        if state is True:
```

Comment:
what does this provide over 
```suggestion
        if original["suspendedDate"] == actual["suspendedDate"] is True:
```